pkg/restore/service_action.go

```diff
@@ -57,6 +57,11 @@ func (a *serviceAction) Execute(obj runtime.Unstructured, restore *api.Restore)
 		delete(spec, "clusterIP")
 	}
 
+	// Since spec.ports is an optional key, we can ignore 'not found' errors. Also assuming it was a string already.
+	if val, _ := collections.GetString(spec, "spec.ports"); val != "None" {
+		delete(spec, "spec.ports")
```

Comment:
I don't think we want to delete `spec.ports`; instead, down below where we access `spec.ports`, we want to be ok with the field not being there for `spec.type == ExternalName` services.

Reply:
Ah, ok! Makes sense 👍